prov/rxm/src/rxm_conn.c

```diff
@@ -663,24 +727,14 @@ static int rxm_cmap_cm_thread_close(struct rxm_cmap *cmap)
 	return 0;
 }
 
-static int rxm_conn_cleanup(void *arg)
-{
-	return rxm_conn_process_eq_events(container_of(arg, struct rxm_ep,
-						       util_ep));
-}
-
-
 void rxm_cmap_free(struct rxm_cmap *cmap)
 {
 	struct rxm_cmap_peer *peer;
 	struct dlist_entry *entry;
 	size_t i;
 
-	rxm_cmap_cm_thread_close(cmap);
-
-	/* cleanup function would be used in manual progress mode */
-	if (cmap->ep->domain->data_progress != FI_PROGRESS_AUTO)
-		rxm_conn_cleanup(cmap->ep);
+	if (cmap->ep->domain->data_progress == FI_PROGRESS_AUTO)
```

Comment:
Can you move this check inside thread_close and abort if true?

Reply:
Sure.